modules/frontend/combiner/search_tag_values.go

```diff
@@ -0,0 +1,66 @@
+package combiner
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/util"
+)
+
+var (
+	_ Combiner = (*genericCombiner[*tempopb.SearchTagValuesResponse])(nil)
+	_ Combiner = (*genericCombiner[*tempopb.SearchTagValuesV2Response])(nil)
+)
+
+func NewSearchTagValues() Combiner {
+	// Distinct collector with no limit
+	d := util.NewDistinctValueCollector(0, func(_ string) int { return 0 })
+
+	return &genericCombiner[*tempopb.SearchTagValuesResponse]{
+		code:  200,
+		final: &tempopb.SearchTagValuesResponse{TagValues: make([]string, 0)},
+		combine: func(body io.ReadCloser, final *tempopb.SearchTagValuesResponse) error {
+			response := &tempopb.SearchTagValuesResponse{}
+			if err := jsonpb.Unmarshal(body, response); err != nil {
+				return fmt.Errorf("error unmarshalling response body: %w", err)
+			}
+			for _, v := range response.TagValues {
+				d.Collect(v)
+			}
+			return nil
+		},
+		result: func(response *tempopb.SearchTagValuesResponse) (string, error) {
+			response.TagValues = d.Values()
+			return new(jsonpb.Marshaler).MarshalToString(response)
+		},
+	}
+}
+
+func NewSearchTagValuesV2() Combiner {
+	// Distinct collector with no limit
+	d := util.NewDistinctValueCollector(0, func(_ tempopb.TagValue) int { return 0 })
```

Comment:
Does it make sense to impose a limit?  The limits are per-tenant so the total result could be as high as sum(tenant limits).  That seems likely to cause some issues eventually, because the limits exist not just for Tempo but downstream apps too (i.e. Grafana).  Using max(tenant limits) is one possibility but doesn't seem right.

Reply:
my expectation as a user here is that each tenant's limits will be respect, and if I am making a query across two tenants, I expect the limits to be sum of both tenant's limits.

the comment is slightly misleading, we do have a limit here and that is sum of all tenant's limits.

I can make it max(tenant limits) but as you said, that doesn't seem right to me as well.

Reply:
going to leave this as is, will revisit this if it becomes an issue.